Index: scm/skk-custom.scm
===================================================================
--- scm/skk-custom.scm	(revision 5027)
+++ scm/skk-custom.scm	(working copy)
@@ -386,8 +386,7 @@
 	 (lambda ()
 	   skk-use-skkserv?))
 
-(define-custom 'skk-dic-file-name (string-append (sys-datadir)
-						 "/skk/SKK-JISYO.L")
+(define-custom 'skk-dic-file-name "/Library/Dictionaries/SKK/SKK-JISYO.L"
   '(skk-dict dict-files)
   '(pathname regular-file)
   (N_ "System dictionary file")
Index: scm/tutcode-key-custom.scm
===================================================================
--- scm/tutcode-key-custom.scm	(revision 5027)
+++ scm/tutcode-key-custom.scm	(working copy)
@@ -53,7 +53,7 @@
 	       (N_ "[TUT-Code] off")
 	       (N_ "long description will be here"))
 
-(define-custom 'tutcode-kana-toggle-key '("<IgnoreShift>'")
+(define-custom 'tutcode-kana-toggle-key '("<IgnoreShift>'" "<Shift>Private2")
                '(tutcode-keys1 mode-transition)
 	       '(key)
 	       (N_ "[TUT-Code] toggle hiragana/katakana mode")
Index: scm/generic-key-custom.scm
===================================================================
--- scm/generic-key-custom.scm	(revision 5027)
+++ scm/generic-key-custom.scm	(working copy)
@@ -41,13 +41,13 @@
 		     (N_ "long description will be here."))
 
 
-(define-custom 'generic-on-key '("zenkaku-hankaku" "<Shift> ")
+(define-custom 'generic-on-key '("zenkaku-hankaku" "<Shift> " "Private2")
                '(global-keys1)
 	       '(key)
 	       (N_ "[Global] on")
 	       (N_ "long description will be here"))
 
-(define-custom 'generic-off-key '("zenkaku-hankaku" "<Shift> ")
+(define-custom 'generic-off-key '("zenkaku-hankaku" "<Shift> " "Private1")
                '(global-keys1)
 	       '(key)
 	       (N_ "[Global] off")
Index: uim/uim-helper.c
===================================================================
--- uim/uim-helper.c	(revision 5027)
+++ uim/uim-helper.c	(working copy)
@@ -48,6 +48,8 @@
 #include <errno.h>
 #include <sys/stat.h>
 
+#include <libkern/OSByteOrder.h>
+
 #include "uim-internal.h"
 #include "uim-helper.h"
 #include "uim-util.h"
@@ -205,6 +207,9 @@
     perror("getpeereid failed");
     return -1;
   }
+  if (euid > INT32_MAX)
+    euid = OSSwapInt32(euid);
+
   if ((euid != 0) && (euid != getuid())) {
     fprintf(stderr, "uid mismatch\n");
     return -1;
Index: uim.pc.in
===================================================================
--- uim.pc.in	(revision 5027)
+++ uim.pc.in	(working copy)
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-includedir=@includedir@
+includedir=@prefix@/Headers
 libdir=@libdir@
 datarootdir=@datarootdir@
 datadir=@datadir@
@@ -10,5 +10,5 @@
 Name: uim
 Description: multilingual input method library
 Version: @VERSION@
-Cflags: -I${includedir}/uim
-Libs: -L${libdir} -luim-scm -luim @LIBINTL@ @LIBICONV@
+Cflags: -I${includedir}
+Libs: -framework UIM
Index: scm/action.scm
===================================================================
--- scm/action.scm	(revision 5889)
+++ scm/action.scm	(working copy)
@@ -232,19 +232,23 @@
 
 (define widget-update-configuration!
   (lambda (widget)
-    (let* ((new-config (widget-configuration widget))
-	   (config-updated? (not (equal? (widget-prev-config widget)
-					 new-config))))
-      (widget-set-prev-config! widget new-config)
-      config-updated?)))
+    (if (symbol? widget)
+      #f
+      (let* ((new-config (widget-configuration widget))
+	     (config-updated? (not (equal? (widget-prev-config widget)
+					   new-config))))
+        (widget-set-prev-config! widget new-config)
+        config-updated?))))
 
 (define widget-update-state!
   (lambda (widget)
-    (let* ((new-state (widget-state widget))
-	   (state-updated? (not (equal? (widget-prev-state widget)
-					new-state))))
-      (widget-set-prev-state! widget new-state)
-      state-updated?)))
+    (if (symbol? widget)
+      #f
+      (let* ((new-state (widget-state widget))
+	     (state-updated? (not (equal? (widget-prev-state widget)
+					  new-state))))
+        (widget-set-prev-state! widget new-state)
+        state-updated?))))
 
 (define widget-debug-message
   (lambda (widget location defect)
Index: scm/im-custom.scm
===================================================================
--- scm/im-custom.scm	(revision 5896)
+++ scm/im-custom.scm	(working copy)
@@ -599,7 +599,7 @@
 		     (N_ "Notify")
 		     (N_ "long description will be here."))
 
-(define-custom 'notify-agent 'stderr
+(define-custom 'notify-agent 'growl
   '(notify)
   `(choice
     ,@(uim-notify-get-plugins))
Index: fep/Makefile.am
===================================================================
--- fep/Makefile.am	(revision 5896)
+++ fep/Makefile.am	(working copy)
@@ -9,7 +9,8 @@
 uim_fep_CPPFLAGS = -I$(top_srcdir)
 uim_fep_CFLAGS =
 uim_fep_LDADD = $(top_builddir)/uim/libuim-scm.la \
-		$(top_builddir)/uim/libuim.la @FEP_LIBADD@
+		$(top_builddir)/uim/libuim.la @FEP_LIBADD@ \
+		$(top_builddir)/replace/libreplace.la
 
 uim_fep_tick_SOURCES = uim-fep-tick.c uim-fep.h udsock.c udsock.h
 uim_fep_tick_CPPFLAGS= -I$(top_srcdir)
Index: scm/prime.scm
===================================================================
--- scm/prime.scm	(revision 5896)
+++ scm/prime.scm	(working copy)
@@ -838,7 +838,7 @@
       (or fds
           (begin
             (unlink socket-path)
-            (process-with-daemon "prime" (list "prime" "-u" socket-path))
+            (process-with-daemon "/Library/PRIME/bin/prime" (list "/Library/PRIME/bin/prime" "-u" socket-path))
             (let loop ((fds (prime-open-unix-domain-socket socket-path))
                        (giveup 100))
               (cond ((= giveup 0)
@@ -899,7 +899,7 @@
                      ((eq? prime-server-setting? 'tcpserver)
                       (prime-open-with-tcp-socket prime-tcpserver-name prime-tcpserver-port))
                      ((eq? prime-server-setting? 'pipe)
-                      (prime-open-with-pipe "prime"))
+                      (prime-open-with-pipe "/Library/PRIME/bin/prime"))
                      (else
                       (uim-notify-fatal (N_ "Prime connection is not defined"))))))
       (cons (open-file-port (car fds))
