Index: gui/qt_libraries.gypi
===================================================================
--- gui/qt_libraries.gypi	(revision 30)
+++ gui/qt_libraries.gypi	(working copy)
@@ -38,10 +38,13 @@
     ['OS=="mac"', {
       'link_settings': {
         'libraries': [
-          'libQtCore.dylib',
-          'libQtGui.dylib',
+          'QtCore.framework',
+          'QtGui.framework',
           '$(SDKROOT)/System/Library/Frameworks/Carbon.framework',
         ],
+	'mac_framework_dirs': [
+	  '$(QTDIR)/lib',
+	],
       },
     }],
     ['OS=="linux"', {
Index: gui/tool/mozc_tool_main.cc
===================================================================
--- gui/tool/mozc_tool_main.cc	(revision 30)
+++ gui/tool/mozc_tool_main.cc	(working copy)
@@ -37,7 +37,7 @@
 #include "base/run_level.h"
 #include "base/winmain.h"
 
-DEFINE_string(mode, "about_dialog", "mozc_tool mode");
+DEFINE_string(mode, "config_dialog", "mozc_tool mode");
 
 // Run* are defiend in each qt module
 int RunAboutDialog(int argc, char *argv[]);
