Index: gui/base/locale_util.cc
===================================================================
--- gui/base/locale_util.cc	(revision 30)
+++ gui/base/locale_util.cc	(working copy)
@@ -50,6 +50,7 @@
 
 #include <QtCore/QTextCodec>
 #include <QtGui/QApplication>
+#undef check
 #include <QtGui/QtGui>
 #include <map>
 #include <string>
Index: gyp/common.gypi
===================================================================
--- gyp/common.gypi	(revision 30)
+++ gyp/common.gypi	(working copy)
@@ -50,7 +50,7 @@
     # warning_cflags_cflags will be shared with Mac and Linux.
     'warning_cflags': [
       '-Wall',
-      '-Werror',
+      #'-Werror',
       '-Wno-char-subscripts',
       '-Wno-sign-compare',
       '-Wwrite-strings',
Index: mac/shared_subpackage_info
===================================================================
--- mac/shared_subpackage_info	(revision 0)
+++ mac/shared_subpackage_info	(revision 0)
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+ <key>CFBundleDevelopmentRegion</key>
+ <string>Japanese</string>
+ <key>CFBundleDisplayName</key>
+ <string>MozcTool</string>
+ <key>CFBundleExecutable</key>
+ <string>MozcTool</string>
+ <key>CFBundleIconFile</key>
+ <string>product_icon.icns</string>
+ <key>CFBundleIdentifier</key>
+ <string>org.mozc.inputmethod.Japanese.Tool</string>
+ <key>CFBundleInfoDictionaryVersion</key>
+ <string>6.0</string>
+ <key>CFBundleName</key>
+ <string>MozcTool</string>
+ <key>CFBundlePackageType</key>
+ <string>APPL</string>
+ <key>CFBundleSignature</key>
+ <string>????</string>
+ <key>CFBundleVersion</key>
+ <string>${GOOGLE_VERSIONINFO_LONG}</string>
+ <key>CFBundleShortVersionString</key>
+ <string>${GOOGLE_VERSIONINFO_SHORT}</string>
+ <key>CFBundleGetInfoString</key>
+ <string>${GOOGLE_VERSIONINFO_FINDER}</string>
+ <key>NSHumanReadableCopyright</key>
+ <string>${GOOGLE_VERSIONINFO_ABOUT}</string>
+ <key>CSResourcesFileMapped</key>
+ <true/>
+ <key>NSPrincipalClass</key>
+ <string>NSApplication</string>
+ <key>LSUIElement</key>
+ <string>1</string>
+</dict>
+</plist>
