Index: gui/base/locale_util.cc
===================================================================
--- gui/base/locale_util.cc	(revision 30)
+++ gui/base/locale_util.cc	(working copy)
@@ -50,6 +50,7 @@
 
 #include <QtCore/QTextCodec>
 #include <QtGui/QApplication>
+#undef check
 #include <QtGui/QtGui>
 #include <map>
 #include <string>
Index: gyp/breakpad_mac.gypi
===================================================================
--- gyp/breakpad_mac.gypi	(revision 52)
+++ gyp/breakpad_mac.gypi	(working copy)
@@ -32,9 +32,9 @@
     'mac_breakpad_dir': '<(DEPTH)/../mac/Releases/GoogleBreakpad/',
     'mac_breakpad_framework': '<(mac_breakpad_dir)/GoogleBreakpad.framework',
   },
-  'link_settings': {
-    'libraries': [
-      '<(mac_breakpad_framework)',
-    ],
-  },
+#  'link_settings': {
+#    'libraries': [
+#      '<(mac_breakpad_framework)',
+#    ],
+#  },
 }
Index: gyp/common.gypi
===================================================================
--- gyp/common.gypi	(revision 55)
+++ gyp/common.gypi	(working copy)
@@ -111,7 +111,7 @@
 
     # use_libzinnia represents if zinnia library is used or not.
     # This option is only for Linux.
-    'use_libzinnia%': 1,
+    'use_libzinnia%': 0,
 
     # a flag whether the current build is dev-channel or not.
     'channel_dev%': '0',
