Index: gui/base/locale_util.cc
===================================================================
--- gui/base/locale_util.cc	(revision 30)
+++ gui/base/locale_util.cc	(working copy)
@@ -50,6 +50,7 @@
 
 #include <QtCore/QTextCodec>
 #include <QtGui/QApplication>
+#undef check
 #include <QtGui/QtGui>
 #include <map>
 #include <string>
diff --git gyp/postbuilds_mac.gypi gyp/postbuilds_mac.gypi
index c9400a0..d0777a2 100644
--- gyp/postbuilds_mac.gypi
+++ gyp/postbuilds_mac.gypi
@@ -36,15 +36,15 @@
   },
   'conditions': [
     ['branding=="GoogleJapaneseInput"', {
-      'link_settings': {
-        'libraries': [
-          '<(mac_breakpad_framework)',
-        ],
-      },
+      #'link_settings': {
+      #  'libraries': [
+      #    '<(mac_breakpad_framework)',
+      #  ],
+      #},
       'copies': [
         {
           'files': [
-            '<(mac_breakpad_framework)',
+            #'<(mac_breakpad_framework)',
             '<@(copying_frameworks)',
           ],
           'destination': '<(PRODUCT_DIR)/<(product_name).app/Contents/Frameworks',
