Index: protobuf/protobuf.gyp
===================================================================
--- protobuf/protobuf.gyp	(revision 59)
+++ protobuf/protobuf.gyp	(working copy)
@@ -102,7 +102,6 @@
       'files/src/google/protobuf/compiler/java/java_message_field.cc',
       'files/src/google/protobuf/compiler/java/java_primitive_field.cc',
       'files/src/google/protobuf/compiler/java/java_service.cc',
-      'files/src/google/protobuf/compiler/java/java_string_field.cc',
       'files/src/google/protobuf/compiler/main.cc',
       'files/src/google/protobuf/compiler/plugin.cc',
       'files/src/google/protobuf/compiler/plugin.pb.cc',
