Index: gui/qt_libraries.gypi
===================================================================
--- gui/qt_libraries.gypi	(revision 30)
+++ gui/qt_libraries.gypi	(working copy)
@@ -38,10 +38,13 @@
     ['OS=="mac"', {
       'link_settings': {
         'libraries': [
-          'libQtCore.dylib',
-          'libQtGui.dylib',
+          'QtCore.framework',
+          'QtGui.framework',
           '$(SDKROOT)/System/Library/Frameworks/Carbon.framework',
         ],
+	'mac_framework_dirs': [
+	  '$(QTDIR)/lib',
+	],
       },
     }],
     ['OS=="linux"', {
Index: gui/mac_gui.gypi
===================================================================
--- gui/mac_gui.gypi	(revision 56)
+++ gui/mac_gui.gypi	(working copy)
@@ -44,7 +44,7 @@
           'postbuild_name': 'create symbolic link to frameworks',
           'action': [
             '/bin/ln', '-fs',
-            '/Library/Input Methods/<(branding).app/Contents/Resources/<(branding)Tool.app/Contents/Frameworks',
+            '/Library/Mozc/<(branding)Tool.app/Contents/Frameworks',
             '${BUILT_PRODUCTS_DIR}/<(product_name).app/Contents',
           ],
         },
Index: gui/tool/mozc_tool_libmain.cc
===================================================================
--- gui/tool/mozc_tool_libmain.cc	(revision 55)
+++ gui/tool/mozc_tool_libmain.cc	(working copy)
@@ -40,7 +40,7 @@
 #include "gui/base/debug_util.h"
 #include "base/run_level.h"
 
-DEFINE_string(mode, "about_dialog", "mozc_tool mode");
+DEFINE_string(mode, "config_dialog", "mozc_tool mode");
 
 // Run* are defiend in each qt module
 int RunAboutDialog(int argc, char *argv[]);
Index: gui/character_pad/hand_writing_canvas.cc
===================================================================
--- gui/character_pad/hand_writing_canvas.cc	(revision 56)
+++ gui/character_pad/hand_writing_canvas.cc	(working copy)
@@ -53,7 +53,7 @@
   // TODO(komatsu): Fix the file name to "handwriting-ja.model" like the
   // Windows implementation regardless which data file is actually
   // used.  See also gui.gyp:hand_writing_mac.
-  const char kModelFile[] = "handwriting-light-ja.model";
+  const char kModelFile[] = "MozcTool.app/Contents/Resources/handwriting-ja.model";
   return Util::JoinPath(MacUtil::GetResourcesDirectory(), kModelFile);
 #elif defined(USE_LIBZINNIA)
   // On Linux, use the model for tegaki-zinnia.
