Index: gyp/common.gypi
===================================================================
--- gyp/common.gypi	(revision 39)
+++ gyp/common.gypi	(working copy)
@@ -107,7 +107,7 @@
 
     # use_libprotobuf represents if protobuf library is used or not.
     # This option is only for Linux.
-    'use_libprotobuf%': 1,
+    'use_libprotobuf%': 0,
 
     # a flag whether the current build is dev-channel or not.
     'channel_dev%': '0',
@@ -423,6 +423,9 @@
         'defines': [
           'OS_LINUX',
         ],
+        'include_dirs': [
+          '../protobuf/files/src',
+	],
         'cflags': [
           '<@(gcc_cflags)',
           '-fPIC',
Index: protobuf/genproto.gypi
===================================================================
--- protobuf/genproto.gypi	(revision 39)
+++ protobuf/genproto.gypi	(working copy)
@@ -39,7 +39,7 @@
         '<(protobuf_dir)/protobuf.gyp:install_protoc',
       ],
     }],
-    ['OS!="linux"', {
+    ['OS!="hoge"', {
       'variables': {
         'protoc_command%': '<(relative_dir)/<(mozc_build_tools_dir)/protoc<(EXECUTABLE_SUFFIX)',
       },
Index: protobuf/protobuf.gyp
===================================================================
--- protobuf/protobuf.gyp	(revision 39)
+++ protobuf/protobuf.gyp	(working copy)
@@ -193,7 +193,7 @@
       },
       'conditions': [
         # use system-installed protoc on Linux
-        ['OS!="linux"', {
+        ['OS!="hoge"', {
           'includes' : [
             '../gyp/install_build_tool.gypi',
           ],
