Index: languages/global_language_spec.cc
===================================================================
--- languages/global_language_spec.cc	(revision 91)
+++ languages/global_language_spec.cc	(working copy)
@@ -32,9 +32,7 @@
 namespace mozc {
 namespace language {
 
-namespace {
-LanguageDependentSpecInterface *g_language_dependent_spec = NULL;
-}  // namespace
+LanguageDependentSpecInterface *g_language_dependent_spec;
 
 const LanguageDependentSpecInterface
     *GlobalLanguageSpec::GetLanguageDependentSpec() {
Index: languages/global_language_spec.h
===================================================================
--- languages/global_language_spec.h	(revision 91)
+++ languages/global_language_spec.h	(working copy)
@@ -36,6 +36,8 @@
 namespace mozc {
 namespace language {
 
+extern LanguageDependentSpecInterface *g_language_dependent_spec;
+
 class GlobalLanguageSpec {
  public:
   static const LanguageDependentSpecInterface *GetLanguageDependentSpec();
