Index: session/internal/keymap.cc
===================================================================
--- session/internal/keymap.cc	(revision 103)
+++ session/internal/keymap.cc	(working copy)
@@ -586,6 +586,8 @@
                              CompositionState::TRANSLATE_HALF_ASCII);
   RegisterCompositionCommand("ToggleAlphanumericMode",
                              CompositionState::TOGGLE_ALPHANUMERIC_MODE);
+  RegisterCompositionCommand("KillLine",
+                             CompositionState::KILL_LINE);
   if (kInputModeXCommandSupported) {
     RegisterCompositionCommand("InputModeHiragana",
                                CompositionState::INPUT_MODE_HIRAGANA);
Index: session/internal/keymap_interface.h
===================================================================
--- session/internal/keymap_interface.h	(revision 52)
+++ session/internal/keymap_interface.h	(working copy)
@@ -131,6 +131,7 @@
   INPUT_MODE_FULL_ALPHANUMERIC,
   INPUT_MODE_HALF_ALPHANUMERIC,
   ABORT,  // Abort the server.  The process is killed.
+  KILL_LINE, // kill-line
 };
 };
 
Index: session/session.cc
===================================================================
--- session/session.cc	(revision 52)
+++ session/session.cc	(working copy)
@@ -652,6 +652,9 @@
     case keymap::CompositionState::DEL:
       return Delete(command);
 
+    case keymap::CompositionState::KILL_LINE:
+      return KillLine(command);
+
     case keymap::CompositionState::BACKSPACE:
       return Backspace(command);
 
@@ -1820,6 +1823,25 @@
   return true;
 }
 
+bool Session::KillLine(commands::Command *command) {
+  command->mutable_output()->set_consumed(true);
+  ClearUndoContext();
+  context_->mutable_composer()->KillLine();
+  if (context_->mutable_composer()->Empty()) {
+    // Input/output modes remain.	  
+    context_->mutable_composer()->EditErase();
+    SetSessionState(ImeContext::PRECOMPOSITION, context_.get());
+    OutputMode(command);
+  } else if (context_->mutable_converter()->Suggest(context_->composer())) {
+    DCHECK(context_->converter().IsActive());
+    Output(command);
+    return true;
+  } else {
+    OutputComposition(command);
+  }
+  return true;
+}
+
 bool Session::Backspace(commands::Command *command) {
   command->mutable_output()->set_consumed(true);
   ClearUndoContext();
Index: session/session.h
===================================================================
--- session/session.h	(revision 151)
+++ session/session.h	(working copy)
@@ -129,6 +129,7 @@
   bool InsertSpaceFullWidth(mozc::commands::Command *command);
   bool InsertCharacter(mozc::commands::Command *command);
   bool Delete(mozc::commands::Command *command);
+  bool KillLine(mozc::commands::Command *command);
   bool Backspace(mozc::commands::Command *command);
   bool EditCancel(mozc::commands::Command *command);
   bool EditCancelAndIMEOff(mozc::commands::Command *command);
Index: gui/config_dialog/keymap_en.ts
===================================================================
--- gui/config_dialog/keymap_en.ts	(revision 52)
+++ gui/config_dialog/keymap_en.ts	(working copy)
@@ -104,6 +104,10 @@
         <translation>Delete (erase character after cursor)</translation>
     </message>
     <message>
+        <source>KillLine</source>
+        <translation>Kill line (erase characters after cursor)</translation>
+    </message>
+    <message>
         <source>DisplayAsFullAlphanumeric</source>
         <translation>Display as full-width alphanumeric</translation>
     </message>
Index: composer/composition_interface.h
===================================================================
--- composer/composition_interface.h	(revision 52)
+++ composer/composition_interface.h	(working copy)
@@ -49,6 +49,7 @@
  public:
   virtual ~CompositionInterface() {}
   virtual size_t DeleteAt(size_t position) = 0;
+  virtual size_t KillLineAt(size_t position) = 0;
   virtual size_t InsertAt(size_t position, const string &input) = 0;
   virtual size_t InsertKeyAndPreeditAt(size_t pos,
                                        const string &key,
Index: composer/composer.cc
===================================================================
--- composer/composer.cc	(revision 52)
+++ composer/composer.cc	(working copy)
@@ -463,6 +463,16 @@
   UpdateInputMode();
 }
 
+void Composer::KillLineAt(size_t pos) {
+  position_ = composition_->KillLineAt(pos);
+  UpdateInputMode();
+}
+
+void Composer::KillLine() {
+  position_ = composition_->KillLineAt(position_);
+  UpdateInputMode();
+}
+
 void Composer::EditErase() {
   composition_->Erase();
   position_ = 0;
Index: composer/internal/composition.cc
===================================================================
--- composer/internal/composition.cc	(revision 52)
+++ composer/internal/composition.cc	(working copy)
@@ -117,6 +117,19 @@
   return new_position;
 }
 
+// Deletes right-hand characters of the composition.
+size_t Composition::KillLineAt(const size_t position) {
+  CharChunkList::iterator chunk_it;
+  MaybeSplitChunkAt(position, &chunk_it);
+  const size_t new_position = GetPosition(Transliterators::LOCAL, chunk_it);
+  if (chunk_it == chunks_.end()) {
+    return new_position;
+  }
+  chunks_.erase(chunk_it, chunks_.end());
+
+  return new_position;
+}
+
 size_t Composition::ConvertPosition(
     const size_t position_from,
     const TransliteratorInterface *transliterator_from,
Index: composer/internal/composition.h
===================================================================
--- composer/internal/composition.h	(revision 124)
+++ composer/internal/composition.h	(working copy)
@@ -54,6 +54,7 @@
   virtual ~Composition();
 
   virtual size_t DeleteAt(size_t position);
+  virtual size_t KillLineAt(size_t position);
   virtual size_t InsertAt(size_t position, const string &input);
   virtual size_t InsertKeyAndPreeditAt(size_t position,
                                        const string &key,
Index: composer/composer.h
===================================================================
--- composer/composer.h	(revision 84)
+++ composer/composer.h	(working copy)
@@ -125,6 +125,7 @@
 
   // Deletes a character at specified position.
   void DeleteAt(size_t pos);
+  void KillLineAt(size_t pos);
   // Delete multiple characters beginning at specified position.
   void DeleteRange(size_t pos, size_t length);
 
@@ -138,6 +139,7 @@
   bool InsertCharacterKeyEvent(const commands::KeyEvent &key);
   void InsertCommandCharacter(const InternalCommand internal_command);
   void Delete();
+  void KillLine();
   void Backspace();
 
   // void Undo();
