Index: gui/mac_gui.gypi
===================================================================
--- gui/mac_gui.gypi	(revision 56)
+++ gui/mac_gui.gypi	(working copy)
@@ -44,7 +44,7 @@
           'postbuild_name': 'create symbolic link to frameworks',
           'action': [
             '/bin/ln', '-fs',
-            '/Library/Input Methods/<(branding).app/Contents/Resources/<(branding)Tool.app/Contents/Frameworks',
+            '/Library/Mozc/<(branding)Tool.app/Contents/Frameworks',
             '${BUILT_PRODUCTS_DIR}/<(product_name).app/Contents',
           ],
         },
Index: gui/tool/mozc_tool_libmain.cc
===================================================================
--- gui/tool/mozc_tool_libmain.cc	(revision 151)
+++ gui/tool/mozc_tool_libmain.cc	(working copy)
@@ -46,7 +46,7 @@
 #include "gui/base/debug_util.h"
 #include "gui/base/win_util.h"
 
-DEFINE_string(mode, "about_dialog", "mozc_tool mode");
+DEFINE_string(mode, "config_dialog", "mozc_tool mode");
 
 // Run* are defiend in each qt module
 int RunAboutDialog(int argc, char *argv[]);
@@ -93,9 +93,11 @@
 #endif  // OS_MACOSX
 
 int RunMozcTool(int argc, char *argv[]) {
+#if 0
   if (mozc::config::StatsConfigUtil::IsEnabled()) {
     mozc::CrashReportHandler::Initialize(false);
   }
+#endif
 #ifdef OS_MACOSX
   // OSX's app won't accept command line flags.
   // Here we read the flags by using --fromenv option
